core/src/hir/lowering.rs

```diff
@@ -716,6 +716,27 @@ impl<'ast> LoweringContext<'ast> {
                     .map(|(lt, m)| Borrow::new(ltl.lower_lifetime(lt), *m)),
                 PrimitiveType::from_ast(*prim),
             ))),
+            ast::TypeName::Function(input_types, out_type) => {
```

Comment:
issue: we should validate that callbacks are allowed in the backend first

just check `!self.attr_validator.attrs_supported().callback` and return an error otherwise, like so


https://github.com/rust-diplomat/diplomat/pull/558/commits/b5a78a7775eef94e7be113e2b77b19e7709c8536

Reply:
good catch!